src/packet_handler.c

```diff
@@ -0,0 +1,50 @@
+#include "packet_handler.h"
+
+
+bool is_match(struct packet *packet, Rule *rule) {
+    if ((!rule->filter.src.ip || rule->filter.src.ip == packet->ip_header->saddr) &&
+        (!rule->filter.dst.ip || rule->filter.dst.ip == packet->ip_header->daddr) &&
+        (!rule->filter.protocol || rule->filter.protocol == packet->ip_header->protocol))
+    {
+        switch (packet->ip_header->protocol)
+        {
+            case IPPROTO_TCP:
+                if ((!rule->filter.src.port || rule->filter.src.port == packet->layer4.tcp_header->source) &&
+                    (!rule->filter.dst.port || rule->filter.dst.port == packet->layer4.tcp_header->dest)) 
+                {
+                    return true;
+                }
+                break;
+            default:
+                break;
+        }
+    }
+    return false;
+}
+
+Rule *get_matching_rule(struct packet *packet, Rule* rules, int rule_count) {
```

Comment:
Be consistent and keep the * near the variable name and not type.
In this case replace `Rule* rules` with `Rule *rules`